params/config_test.go

```diff
@@ -328,3 +328,75 @@ func TestChainConfigMarshalWithUpgrades(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, config, unmarshalled)
 }
+
+func TestChainConfigMarshalWithUpgradesAndOptionalUpgrade(t *testing.T) {
+	config := ChainConfigWithUpgradesJSON{
```

Comment:
This seems to test the json marshal/unmarshal, but does not test the new marshal / unmarshal binary code.

We should add test coverage for the binary marshal / unmarshal as well.